ksqldb-engine/src/main/java/io/confluent/ksql/topic/TopicFactory.java

```diff
@@ -31,17 +32,21 @@
   private TopicFactory() {
   }
 
-  public static KsqlTopic create(final CreateSourceProperties properties) {
+  public static KsqlTopic create(
+      final CreateSourceProperties properties,
+      final KsqlConfig config
+  ) {
     final String kafkaTopicName = properties.getKafkaTopic();
 
     final Optional<WindowType> windowType = properties.getWindowType();
     final Optional<Duration> windowSize = properties.getWindowSize();
 
+    final FormatInfo keyFormatInfo = properties.getKeyFormatInfo()
+        .orElseGet(() -> getDefaultKeyFormat(config));
```

Comment:
This is the code that is responsible for handling `KEY_FORMAT` for C* statements.  You'd need to extend it to handle `FORMAT` too.